cpp/modmesh/universe/polygon.hpp

```diff
@@ -781,6 +788,617 @@ class TrianglePad
+    Polygon3d & operator=(Polygon3d &&) = default;
+    ~Polygon3d() = default;
+
+    size_t polygon_id() const { return m_id; }
+    std::shared_ptr<polygon_pad_type> pad() const { return m_pad; }
+
+    size_t nnode() const
+    {
+        if (m_id >= m_pad->m_begins.size())
+        {
+            throw std::out_of_range(
+                std::format("Polygon3d::nnode: polygon_id {} >= num_polygons {}",
+                            m_id,
+                            m_pad->m_begins.size()));
+        }
+        typename polygon_pad_type::ssize_type const begin_index = m_pad->m_begins[m_id];
+        typename polygon_pad_type::ssize_type const end_index = m_pad->m_ends[m_id];
+        return static_cast<size_t>(end_index - begin_index);
+    }
+
+    uint8_t ndim() const
+    {
+        return m_pad->ndim();
+    }
+
+    point_type node(size_t index) const
+    {
+        if (m_id >= m_pad->m_begins.size())
+        {
+            throw std::out_of_range(
+                std::format("Polygon3d::node: polygon_id {} >= num_polygons {}",
+                            m_id,
+                            m_pad->m_begins.size()));
+        }
+        typename polygon_pad_type::ssize_type const begin_index = m_pad->m_begins[m_id];
+        typename polygon_pad_type::ssize_type const end_index = m_pad->m_ends[m_id];
+        size_t const count = static_cast<size_t>(end_index - begin_index);
+        if (index >= count)
+        {
+            throw std::out_of_range(
+                std::format("Polygon3d::node: index {} >= count {}",
+                            index,
+                            count));
+        }
+        return m_pad->m_points->get(static_cast<size_t>(begin_index + static_cast<typename polygon_pad_type::ssize_type>(index)));
+    }
+
+    segment_type edge(size_t index) const
+    {
+        if (m_id >= m_pad->m_begins.size())
+        {
+            throw std::out_of_range(
+                std::format("Polygon3d::edge: polygon_id {} >= num_polygons {}",
+                            m_id,
+                            m_pad->m_begins.size()));
+        }
+        typename polygon_pad_type::ssize_type const begin_index = m_pad->m_begins[m_id];
+        typename polygon_pad_type::ssize_type const end_index = m_pad->m_ends[m_id];
+        size_t const count = static_cast<size_t>(end_index - begin_index);
+        if (index >= count)
+        {
+            throw std::out_of_range(
+                std::format("Polygon3d::edge: index {} >= count {}",
+                            index,
+                            count));
+        }
+        point_type const p0 = m_pad->m_points->get(static_cast<size_t>(begin_index + static_cast<typename polygon_pad_type::ssize_type>(index)));
+        point_type const p1 = m_pad->m_points->get(static_cast<size_t>(begin_index + static_cast<typename polygon_pad_type::ssize_type>((index + 1) % count)));
+        return segment_type(p0, p1);
+    }
+
+    value_type compute_signed_area() const
```

Comment:
`Polygon3d::compute_signed_area()` is too long and has a loop.  It should be moved outside the class declaration.